.github/workflows/rpk-build.yml

```diff
@@ -8,30 +8,124 @@
 # by the Apache License, Version 2.0
 
 name: Go
-on: [push, pull_request]
+on:
+  push:
+    tags: 'release-*'
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - main
+      - dev
 
 jobs:
   build:
-    name: Build
+    name: Test and Build
     runs-on: ubuntu-latest
     steps:
 
+    - name: Check out code
+      uses: actions/checkout@v2
+
     - name: Set up Go
       uses: actions/setup-go@v2
       with:
-        go-version: ^1.15
+        go-version: 1.13.6
       id: go
 
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '12.16.1'
+
+    - name: Install JS
+      working-directory: src/js
+      run: |
+        npm install
+
+    - name: Generate WASM deps
+      working-directory: src/js
+      run: |
+        npm run generate:serialization
+        npm run build:public
+        sed -i 's/%CONTENT%/"$(cat vectorizedDependency.js)"/g' ../go/rpk/pkg/cli/cmd/wasm/template/vectorized_dependency.go
 
     - name: Test
+      working-directory: src/go/rpk/
       run: go test -v ./...
+
+    - name: Build Linux
```

Comment:
Should the two `Build XXX` steps be described in a matrix `[linux, darwin]`?

Reply:
Oh cool! I'll look into it.